### PR TITLE
Add new replicator role to distinguish follower vs. observer

### PR DESCRIPTION
**Pull request by iprentic**
For 2-ack mode, we need to differentiate between a follower and observer. Add a new replicator role for this.

### CODE REVIEW

**Comment by jaricftw** [comment]
I think we also need to change here: https://github.com/pinterest/rocksplicator/blob/bd3842591ece082ca8502bf307b28e277bee5eba/rocksdb_replicator/rocksdb_replicator.cpp#L126 otherwise the observer won't perform any replication request. 

Can you test this in a private build?

File: rocksdb_replicator/thrift/replicator.thrift [hunk 1]
```diff
@@ -56,7 +59,8 @@ struct Update {
 enum ReplicaRole {
   NOOP = 0, // no replication needed
   FOLLOWER = 1,
-  LEADER = 2
+  LEADER = 2,
+  OBSERVER = 3 // replicates but should not be counted as a follower ack
```
Comment (jaricftw) [inline rocksdb_replicator/thrift/replicator.thrift:63]: can you update https://github.com/pinterest/rocksplicator/blob/master/rocksdb_replicator/utils.h#L21 as well

**Comment by iprentic** [comment]
> I think we also need to change here:
> 
> https://github.com/pinterest/rocksplicator/blob/bd3842591ece082ca8502bf307b28e277bee5eba/rocksdb_replicator/rocksdb_replicator.cpp#L126
> 
> otherwise the observer won't perform any replication request.
> Can you test this in a private build?

Tested on rocksobserver cluster only: Worked
Tested on rocksobserver cluster and rockstorewidcolumn clusters: worked
Tested on rockstorewidecolumn cluster only: worked

File: rocksdb_replicator/utils.h [hunk 2]
```diff
@@ -24,6 +24,8 @@ const char* ReplicaRoleString(ReplicaRole role) {
         return "NOOP";
     case ReplicaRole::FOLLOWER:
         return "FOLLOWER";
+    case ReplicaRole::OBSERVER:
```
Comment (jaricftw) [inline rocksdb_replicator/utils.h:27]: can you add unit tests for this -- I think we already have a test for this function

**Review by jaricftw — APPROVED** [review]
LGTM

**Pull request merged**